modules/client/src/internal/utils.ts

```diff
@@ -824,3 +865,812 @@ export function toPluginPermissionOperationType(
       throw new InvalidPermissionOperationType();
   }
 }
+
+// function that compares 2 generic arrays
+// and returns true if they are equal
+// and false if they are not
+export function compareArrays<T>(array1: T[], array2: T[]): boolean {
+  return JSON.stringify(array1) === JSON.stringify(array2);
+}
+async function getPluginInstallations(
+  daoAddress: string,
+  pluginAddress: string,
+  graphql: IClientGraphQLCore,
+): Promise<SubgraphPluginInstallationListItem[]> {
+  const name = "pluginInstallations";
+  type U = { pluginInstallations: SubgraphPluginInstallationListItem[] };
+  const query = QueryPluginInstallations;
+  const params = {
+    where: {
+      plugin: pluginAddress.toLowerCase(),
+      dao: daoAddress.toLowerCase(),
+    },
+  };
+  const res = await graphql.request<U>({
+    query,
+    params,
+    name,
+  });
+  const { pluginInstallations } = res;
+  return pluginInstallations;
+}
+export async function validateGrantUpdatePluginPermissionAction(
+  action: DaoAction,
+  pspAddress: string,
+  daoAddress: string,
+  graphql: IClientGraphQLCore,
+): Promise<PluginUpdateProposalInValidityCause[]> {
+  const causes: PluginUpdateProposalInValidityCause[] = [];
+  const decodedPermission = decodeGrantAction(action.data);
+  const pluginInstallations = await getPluginInstallations(
+    daoAddress,
+    decodedPermission.where,
+    graphql,
+  );
+  if (pluginInstallations.length === 0) {
+    causes.push(
+      PluginUpdateProposalInValidityCause
+        .INVALID_GRANT_UPDATE_PERMISSION_WHERE_ADDRESS,
+    );
+  }
+  if (action.value.toString() !== "0") {
+    causes.push(
+      PluginUpdateProposalInValidityCause
+        .INVALID_GRANT_UPDATE_PERMISSION_VALUE,
+    );
+  }
+  if (decodedPermission.who !== pspAddress) {
+    causes.push(
+      PluginUpdateProposalInValidityCause
+        .INVALID_GRANT_UPDATE_PERMISSION_WHO_ADDRESS,
+    );
+  }
+  if (
+    decodedPermission.permission !== Permissions.UPGRADE_PLUGIN_PERMISSION
```

Comment:
I tried to look up what is happening here but I couldn't figure it out.

```
type PermissionParamsBase = {
  where: string;
  who: string;
  permission: string;
};
type PermissionDecodedParamsBase = PermissionParamsBase & {
  permissionId: string;
};
```
What is the difference between `permission` and `permissionId` ?

Reply:
Clarified on a call. It is the permission name of which we are doing the keccak256 hash